fineract-provider/src/main/java/org/apache/fineract/infrastructure/bulkimport/service/BulkImportWorkbookPopulatorServiceImpl.java

```diff
@@ -0,0 +1,108 @@
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.bulkimport.service;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.ResponseBuilder;
+
+import org.apache.fineract.infrastructure.bulkimport.populator.WorkbookPopulator;
+import org.apache.fineract.infrastructure.bulkimport.populator.office.OfficeWorkbookPopulator;
+import org.apache.fineract.infrastructure.core.exception.GeneralPlatformDomainRuleException;
+import org.apache.fineract.infrastructure.core.service.DateUtils;
+import org.apache.fineract.infrastructure.security.service.PlatformSecurityContext;
+import org.apache.fineract.organisation.office.api.OfficeApiConstants;
+import org.apache.fineract.organisation.office.data.OfficeData;
+import org.apache.fineract.organisation.office.service.OfficeReadPlatformService;
+
+import org.apache.poi.hssf.usermodel.HSSFWorkbook;
+import org.apache.poi.ss.usermodel.Workbook;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class BulkImportWorkbookPopulatorServiceImpl implements BulkImportWorkbookPopulatorService {
+
+	private final PlatformSecurityContext context;
+	private final OfficeReadPlatformService officeReadPlatformService;
+
+	@Autowired
+	public BulkImportWorkbookPopulatorServiceImpl(final PlatformSecurityContext context,
+			final OfficeReadPlatformService officeReadPlatformService) {
+		this.context = context;
+		this.officeReadPlatformService = officeReadPlatformService;
+	}
+
+	@Override
+	public Response getTemplate(final String entityType, final Long officeId, final Long staffId,final Long centerId,
+			final Long clientId,final Long groupId, final Long productId,final Long fundId,
+			final Long paymentTypeId,final String code,final Long glAccountId) {
+
+		WorkbookPopulator populator = null;
+		final Workbook workbook = new HSSFWorkbook();
+		if (entityType.trim().equalsIgnoreCase(OfficeApiConstants.OFFICE_RESOURCE_NAME)) {
+			populator = populateOfficeWorkbook(officeId);
+		} else
+			throw new GeneralPlatformDomainRuleException("error.msg.unable.to.find.resource",
```

Comment:
please use brackets always ... even if the following instructions is just one line ... this makes it easier to read and understand.